### ISSUE TITLE
Pluggable, inheritable "master" contexts; Subclassable default context

### ISSUE THREAD

**Issue by dag** [dag/attest#28]
```
class FlaskTests(Tests):
    contexts = (app.test_request_context,)
```

Also should be easy to set a default context that for instance yields a `Assert(app.test_client())` so that tests can be written like so:

```
api = FlaskTests()

@api.test
def json(client):
    response = client.get('/api/')
    response.mimetype == 'application/json'
    response.status_code == 200
```

**Comment by dag** [comment]
New idea: Tests.context can register many contexts that are entered in order. Write factory functions for reusability:

```
def flask_tests(app):
    tests = Tests()
    @tests.context
    def context():
        with app.test_request_context():
            yield Assert(app.test_client())
    return tests

api = flask_tests(app)

@api.context
def should_still_work_and_not_override(): pass
```

One problem is how to handle yields from multiple contexts.

**Comment by dag** [comment]
Maybe yields could be passed down:

```
@api.context
def context(client):
    do_custom()
    yield client, maybe_something_more
```

**Comment by dag** [comment]
Nested contexts, closed by eaadee8f7c4fc84e5f8e2f19703dbccaba169fd1
